File: Artist/app/src/main/java/com/example/artist/ui/toptrackslisting/TopTracksView.java
package com.example.artist.ui.toptrackslisting;


import com.example.artist.models.Track;

import java.util.List;


public interface TopTracksView {
    void showProgress();

    void hideProgress();

    void showError();

    void updateData(List<Track> tracks);
    void showEmpty();
    void hidEmpty();
}

File: Artist/app/src/main/java/com/example/artist/ui/toptrackslisting/TopTracksView.kt
package com.example.artist.ui.toptrackslisting

import com.example.artist.models.Track

interface TopTracksView {
    fun showProgress()
    fun hideProgress()
    fun showError()
    fun updateData(tracks: MutableList<Track>)
    fun showEmpty()
    fun hidEmpty()
}